cranelift/interpreter/src/interpreter.rs

```diff
@@ -308,7 +315,12 @@ impl<'a> State<'a, DataValue> for InterpreterState<'a> {
         Address::from_parts(size, AddressRegion::Stack, 0, final_offset)
     }
 
-    fn checked_load(&self, addr: Address, ty: Type) -> Result<DataValue, MemoryError> {
+    fn checked_load(
+        &self,
+        addr: Address,
+        ty: Type,
+        mem_flags: MemFlags,
```

Comment:
For future work, not part of this PR: @afonso360, can/should `checked_load` validate mem-flags like `aligned`, now that they're passed in here anyway? And same question for `checked_store`.

Reply:
Oh yeah it absolutely should! I've opened #5899 for this